### PR TITLE
Support multiple sort directions #118

### PR DESCRIPTION
**Pull request by zaharchenko-evgeny**
fix for Support multiple sort directions jmix-framework/jmix#332

### CODE REVIEW

**Comment by zaharchenko-evgeny** [comment]
Multiple sort example:
GraphQL order query
```
{ orderBy: [{age: ASC}, {name: DESC}] }
```
this GraphQL request should generate following JPQL 
```
... order by e.age asc, e.name desc
```

File: data/src/main/java/io/jmix/data/impl/jpql/generator/SortJpqlGenerator.java [hunk 1]
```diff
@@ -59,170 +59,164 @@ public String processQuery(String entityName, List<String> valueProperties, Stri
...
-        Map<Sort.Direction, List<Sort.Order>> directions = orders.stream()
-                .collect(Collectors.groupingBy(Sort.Order::getDirection));
-        if (directions.size() > 1) {
-            throw new UnsupportedOperationException("Sorting by multiple properties in different directions is not supported");
-        }
-        boolean asc = directions.keySet().iterator().next() == Sort.Direction.ASC;
+        Sort.Direction defaultSort = Sort.Direction.ASC;
 
-        List<String> sortExpressions = new ArrayList<>();
+        Map<String,Sort.Direction> sortExpressions = new LinkedHashMap<>();
```
Comment (gorbunkov) [inline data/src/main/java/io/jmix/data/impl/jpql/generator/SortJpqlGenerator.java:64]: Everywhere in this file and other files use standard formatting, i.e. add a space between String and Sort.Direction in generics list: Map<String, Sort.Direction>

File: eclipselink/src/test/groovy/jpql_sort/QuerySortTest.groovy [hunk 2]
```diff
@@ -148,7 +148,10 @@ class QuerySortTest extends DataSpec {
 
         then:
 
-        thrown(UnsupportedOperationException)
+        queryBuilder.getResultQueryString() == 'select u from sec$User u order by u.login asc, u.name desc, u.id asc'
```
Comment (gorbunkov) [inline eclipselink/src/test/groovy/jpql_sort/QuerySortTest.groovy:151]: The current test should change its name. The "sort by multiple properties in different directions is not supported" is not a valid one any more.

Comment (gorbunkov) [inline eclipselink/src/test/groovy/jpql_sort/QuerySortTest.groovy:151]: getResultQueryString() is called twice in this test. No need to call it in "when" section

Comment (zaharchenko-evgeny) [reply eclipselink/src/test/groovy/jpql_sort/QuerySortTest.groovy:151]: fixed tests

Comment (zaharchenko-evgeny) [reply eclipselink/src/test/groovy/jpql_sort/QuerySortTest.groovy:151]: fixed

File: data/src/main/java/io/jmix/data/impl/jpql/generator/SortJpqlGenerator.java (hunk 1 above)
Comment (zaharchenko-evgeny) [reply data/src/main/java/io/jmix/data/impl/jpql/generator/SortJpqlGenerator.java]: fixed

**Review by gorbunkov — APPROVED** [review]

**Pull request merged**